server/adaptors/integrations/integrations_builder.ts

```diff
@@ -42,12 +44,67 @@ export class IntegrationInstanceBuilder {
       return Promise.reject(assets.error);
     }
     const remapped = this.remapIDs(this.getSavedObjectBundles(assets.value));
-    const withDataSource = this.remapDataSource(remapped, options.dataSource);
-    const refs = await this.postAssets(withDataSource);
+    const withDataSource = this.remapDataSource(remapped, options.indexPattern);
+    const withSubstitutedQueries = this.substituteQueries(
+      withDataSource,
+      options.dataSource,
+      options.tableName
+    );
+    const refs = await this.postAssets(withSubstitutedQueries as SavedObjectsBulkCreateObject[]);
     const builtInstance = await this.buildInstance(integration, refs, options);
     return builtInstance;
   }
 
+  // If we have a data source or table specified, hunt for saved queries and update them with the
+  // new DS/table.
+  substituteQueries(assets: SavedObject[], dataSource?: string, tableName?: string): SavedObject[] {
```

Comment:
Heart of the substitution updates: here's where we search for saved queries to update